File: src/test/java/io/bloco/faker/components/FoodTest.java
package io.bloco.faker.components;

import org.junit.Before;
import org.junit.Test;

import io.bloco.faker.Faker;

import static io.bloco.faker.test_helpers.RegularExpressionMatcher.matchesPattern;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertThat;

public class FoodTest {

    private Faker faker;

    @Before
    public void setUp() {
        faker = new Faker();
    }

    @Test
    public void dish() {
        assertNotNull(faker.food.dish());
    }

    @Test
    public void description() {
        assertNotNull(faker.food.description());
    }

    @Test
    public void ingredient() {
        assertNotNull(faker.food.ingredient());
    }

    @Test
    public void spice() {
        assertNotNull(faker.food.spice());
    }

    @Test
    public void measurement() {
        assertNotNull(faker.food.measurement());
        assertThat(faker.food.measurement(), matchesPattern("[\\d\\/]+ \\w+"));
    }

    @Test
    public void metricMeasurement() {
        assertNotNull(faker.food.metricMeasurement());
    }
}

File: lib/src/test/kotlin/io/bloco/faker/components/FoodTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import junit.framework.TestCase.assertNotNull
import junit.framework.TestCase.assertTrue
import org.junit.Test

class FoodTest {
    private val faker = Faker()

    @Test
    fun dish() {
        assertNotNull(faker.food.dish())
    }

    @Test
    fun description() {
        assertNotNull(faker.food.description())
    }

    @Test
    fun ingredient() {
        assertNotNull(faker.food.ingredient())
    }

    @Test
    fun spice() {
        assertNotNull(faker.food.spice())
    }

    @Test
    fun measurement() {
        assertNotNull(faker.food.measurement())
        assertTrue(faker.food.measurement().matches(Regex("[\\d\\/]+ \\w+")))
    }

    @Test
    fun metricMeasurement() {
        assertNotNull(faker.food.metricMeasurement())
    }
}